### PR TITLE
Sprint preview combined pull request (changes before demo)

### PR DESCRIPTION
**Pull request by Akr1s**

### CODE REVIEW

File: frontend/public/index.html [hunk 1]
```diff
@@ -38,5 +38,6 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
+    <div id="modal-portal"></div>
```
Comment (Akr1s) [inline frontend/public/index.html:41]: Remove later

File: frontend/src/actions/groups.js [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
+export const addGroup = (res) => {
     return {
-        type: actionTypes.SHOW_ALL_GROUPS,
+        type: actionTypes.ADD_GROUP,
         result: res,
     };
 };
 
-export const setDisabledGroups = (res) => {
+export const updateGroupSusses = (res) => {
```
Comment (Akr1s) [inline frontend/src/actions/groups.js:10]: @Niknikson Susses -> Success

File: frontend/src/actions/groups.js [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
+export const showAllGroups = (res) => {
     return {
-        type: actionTypes.ADD_GROUP,
+        type: actionTypes.SHOW_ALL_GROUPS,
         result: res,
     };
 };
 
-export const deleteGroup = (res) => {
+export const deleteGroupSusses = (res) => {
```
Comment (Akr1s) [inline frontend/src/actions/groups.js:24]: @Niknikson as above

File: frontend/src/actions/groups.js [hunk 4]
```diff
@@ -35,13 +35,40 @@ export const selectGroup = (res) => {
...
 
-export const updateGroup = (res) => {
-    return {
-        type: actionTypes.UPDATE_GROUP,
-        result: res,
-    };
-};
-
-export const clearGroup = () => ({
+export const clearGroupSusses = () => ({
```
Comment (Akr1s) [inline frontend/src/actions/groups.js:38]: @Niknikson  as above

File: frontend/src/actions/schedule.js [hunk 5]
```diff
@@ -7,16 +7,99 @@ export const setScheduleItems = (res) => {
...
+        type: actionTypes.GET_ALL_PUBLIC_TEACHERS_BY_DEPARTMENT_START,
+        departmentId,
+    };
+};
+
 export const setCurrentSemester = (res) => {
     return {
         type: actionTypes.SET_CURRENT_SEMESTER,
-        result: res,
+        payload: res,
```
Comment (Akr1s) [inline frontend/src/actions/schedule.js:26]: @Akr1s unify code style

File: frontend/src/actions/schedule.js [hunk 6]
```diff
@@ -7,16 +7,99 @@ export const setScheduleItems = (res) => {
...
+    };
+};
+
 export const setDefaultSemester = (res) => {
     return {
         type: actionTypes.SET_DEFAULT_SEMESTER,
-        result: res,
+        payload: res,
+    };
+};
```
Comment (Akr1s) [inline frontend/src/actions/schedule.js:98]: @Akr1s as above

File: frontend/src/components/ClassForm/ClassForm.scss [hunk 7]
```diff
@@ -1,14 +1,20 @@
...
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    margin: 8px 0;
 
-	.class-input {
-		width: 100% !important;
-	}
+    .class-input {
+        width: 100% !important;
```
Comment (Akr1s) [inline frontend/src/components/ClassForm/ClassForm.scss:8]: Do not use `important`

File: frontend/src/components/GroupPage/GroupPage.scss [hunk 8]
```diff
@@ -30,8 +38,8 @@ $group-size: 10px;
 
   &-card {
     box-sizing: border-box;
-    height: 10*$group-size;
-    width: 12*$group-size;
+    height: 12*$group-size;
+    width: 13*$group-size;
```
Comment (Akr1s) [inline frontend/src/components/GroupPage/GroupPage.scss:42]: Why exactly `12` and `13`?

File: frontend/src/components/GroupSchedulePage/GroupsList.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+            onChange={() => {
+                handleChange('teacher', 0);
+            }}
+        >
+            <option className="option-item" value={0} />
+            {groups.map((group) => (
+                <option key={group.id} value={group.id} className="option-item">
+                    {group.title}
+                </option>
+            ))}
```
Comment (Akr1s) [inline frontend/src/components/GroupSchedulePage/GroupsList.js:29]: @Akr1s  destructure `group`

File: frontend/src/components/GroupSchedulePage/SemestersList.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+                component={SelectField}
+                label={t(FORM_SEMESTER_LABEL)}
+                type="text"
+                onChange={(e) => setSemesterId(e.target.value)}
+            >
+                {semesters.map((semester) => (
+                    <option key={semester.id} value={semester.id} className="option-item">
+                        {semester.description}
+                    </option>
+                ))}
```
Comment (Akr1s) [inline frontend/src/components/GroupSchedulePage/SemestersList.js:30]: @Akr1s  destructure `semester`

File: frontend/src/components/GroupSidebar/GroupSidebar.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    const {
+        createGroupStart,
+        updateGroupStart,
+        clearGroupStart,
+        setIsDisabled,
+        isDisabled,
+        setTerm,
+    } = props;
+
+    const SearchChange = setTerm;
```
Comment (Akr1s) [inline frontend/src/components/GroupSidebar/GroupSidebar.js:15]: @Niknikson why should we reassign this?

File: frontend/src/components/GroupSidebar/GroupSidebar.js [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+    const onSubmitGroupForm = (data) => {
+        return !data.id ? createGroupStart(data) : updateGroupStart(data);
+    };
+
+    return (
+        <aside className="search-list__panel">
+            <SearchPanel
+                SearchChange={SearchChange}
+                showDisabled={() => setIsDisabled((prev) => !prev)}
```
Comment (Akr1s) [inline frontend/src/components/GroupSidebar/GroupSidebar.js:25]: @Niknikson we can call this method toggleDisabled

File: frontend/src/components/GroupSidebar/GroupSidebar.js [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+            />
+            {!isDisabled && (
+                <AddGroup
+                    className="form"
+                    onSubmit={onSubmitGroupForm}
+                    onReset={() => clearGroupStart()}
```
Comment (Akr1s) [inline frontend/src/components/GroupSidebar/GroupSidebar.js:31]: @Niknikson arrow function wrapper is not necessary here

File: frontend/src/components/LessonsPage/LessonPage.js [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+        setIsOpenCopyLessonDialog(false);
+        if (!isNil(group)) {
+            copyLessonCard(group, lesson);
+        }
+    };
+
+    const submitCopySemester = (values) => {
+        const toSemesterId = currentSemester.id;
+        const fromSemesterId = +values.fromSemesterId;
+        CopyLessonsFromSemesterService({ ...values, toSemesterId, fromSemesterId });
```
Comment (Akr1s) [inline frontend/src/components/LessonsPage/LessonPage.js:121]: @Yashchemskyi-Oleksandr rewrite the service to saga

File: frontend/src/components/LessonsPage/Lessons.js [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+        visibleItems,
+        onClickOpen,
+        onCopyLesson,
+        groupId,
+        groups,
+        loading,
+        selectLessonCardOf,
+    } = props;
+
+    const searchTitleGroupByID = (id) => groups.find((group) => group.id === +id)?.title;
```
Comment (Akr1s) [inline frontend/src/components/LessonsPage/Lessons.js:20]: @Yashchemskyi-Oleksandr  can we split these two operations for readability?

File: frontend/src/components/LessonsPage/Search/Search.js [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+                )}
+                <span className="group-lesson">
+                    <h1 className="lesson-page-h">{t(LESSON_FOR_GROUP_TITLE)}</h1>
+                    <Autocomplete
+                        id="group"
+                        clearOnEscape
+                        openOnFocus
+                        options={groups}
+                        getOptionLabel={(option) => option.title}
+                        onChange={(event, newValue) => {
```
Comment (Akr1s) [inline frontend/src/components/LessonsPage/Search/Search.js:40]: @Yashchemskyi-Oleksandr  replace unused `event` with `_`

File: frontend/src/components/Semester/SemesterForm/SemesterForm.scss [hunk 17]
```diff
@@ -33,14 +33,14 @@
...
-.group-options{
+.group-options {
     display: flex;
     flex-direction: column;
     flex-wrap: wrap;
     align-items: center;
     align-content: center;
 
-    Button{
+    Button {
```
Comment (Akr1s) [inline frontend/src/components/Semester/SemesterForm/SemesterForm.scss:43]: what is this selector?

File: frontend/src/components/Semester/SemesterItem.scss [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+        text-align: center;
+    }
+}
+
+.wrapper {
+    width: 100%;
+}
+
+.current {
+    box-shadow: 0 0px 15px 1px $good-color;
```
Comment (Akr1s) [inline frontend/src/components/Semester/SemesterItem.scss:32]: @ViktoriiaKosovan  what is `good-color`?

File: frontend/src/services/groupService.js [hunk 19]
```diff
@@ -70,14 +23,18 @@ export const showAllGroupsService = () => {
...
     return axios
         .put(GROUP_URL, data)
         .then((response) => {
-            store.dispatch(updateGroup(response.data));
+            store.dispatch(updateGroupSusses(response.data));
             selectGroupService(null);
-            getDisabledGroupsService();
-            showAllGroupsService();
+            // getDisabledGroupsService();
+            // showAllGroupsService();
```
Comment (Akr1s) [inline frontend/src/services/groupService.js:37]: replace `dispatch` with `mapDispatchToProps`

File: frontend/src/components/AddStudentForm/AddStudentForm.js [hunk 20]
```diff
@@ -0,0 +1,120 @@
...
+        <Card additionClassName="form-card teacher-form">
+            <form className="createTeacherForm w-100" onSubmit={handleSubmit}>
+                <Field
+                    className="form-field"
+                    name="surname"
+                    id="surname"
+                    component={renderTextField}
+                    type="text"
+                    placeholder={t('surname_placeholder')}
+                    label={t('surname_placeholder')}
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/AddStudentForm/AddStudentForm.js:45]: @Niknikson add to translation constants

File: frontend/src/components/AddStudentForm/AddStudentForm.js [hunk 21]
```diff
@@ -0,0 +1,120 @@
...
+
+                <div className="form-buttons-container">
+                    <Button
+                        className="buttons-style"
+                        variant="contained"
+                        color="primary"
+                        disabled={pristine || submitting}
+                        type="submit"
+                    >
+                        {t('save_button_label')}
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/AddStudentForm/AddStudentForm.js:106]: @Niknikson as above

File: frontend/src/components/AddStudentForm/AddStudentForm.js [hunk 22]
```diff
@@ -0,0 +1,120 @@
...
+                    </Button>
+                    <Button
+                        className="buttons-style"
+                        variant="contained"
+                        disabled={pristine || submitting}
+                        onClick={reset}
+                    >
+                        {student ? t('cancel_button_label') : t('clear_button_label')}
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/AddStudentForm/AddStudentForm.js:114]: @Niknikson as above

File: frontend/src/components/AddStudentForm/AddStudentForm.js [hunk 23]
```diff
@@ -0,0 +1,120 @@
...
+                />
+
+                <Field
+                    className="form-field"
+                    name="name"
+                    id="name"
+                    component={renderTextField}
+                    type="text"
+                    placeholder={t('name_placeholder')}
+                    label={t('name_placeholder')}
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/AddStudentForm/AddStudentForm.js:56]: @Niknikson as above

File: frontend/src/components/AddStudentForm/AddStudentForm.js [hunk 24]
```diff
@@ -0,0 +1,120 @@
...
+                />
+
+                <Field
+                    className="form-field"
+                    name="patronymic"
+                    id="patronymic"
+                    component={renderTextField}
+                    type="text"
+                    placeholder={t('patronymic_placeholder')}
+                    label={t('patronymic_placeholder')}
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/AddStudentForm/AddStudentForm.js:67]: @Niknikson as above

File: frontend/src/components/AddStudentForm/AddStudentForm.js [hunk 25]
```diff
@@ -0,0 +1,120 @@
...
+                />
+
+                <Field
+                    className="form-field"
+                    name="email"
+                    id="email"
+                    component={renderTextField}
+                    type="email"
+                    placeholder={t('email_field')}
+                    label={t('email_field')}
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/AddStudentForm/AddStudentForm.js:78]: @Niknikson as above

File: frontend/src/components/GroupList/GroupList.js [hunk 26]
```diff
@@ -0,0 +1,128 @@
...
+        }
+    }, [isDisabled]);
+
+    const visibleGroups = search(groups, term, ['title']);
+
+    const showConfirmDialog = (currentId, disabledStatus) => {
+        setGroupId(currentId);
+        setConfirmDialogType(disabledStatus);
+        setIsOpenConfirmDialog(true);
+        return undefined;
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/GroupList/GroupList.js:54]: @Niknikson Do you need to return 'undefined' in this case?

File: frontend/src/components/GroupPage/GroupPage.js [hunk 27]
```diff
@@ -0,0 +1,37 @@
+import './GroupPage.scss';
+import React, { useState } from 'react';
+import GroupList from '../GroupList/GroupList';
+import GroupSidebar from '../GroupSidebar/GroupSidebar';
+import SnackbarComponent from '../../share/Snackbar/SnackbarComponent';
+import { handleSnackbarCloseService } from '../../services/snackbarService';
+
+const GroupPage = (props) => {
+    const { isSnackbarOpen, snackbarMessage, snackbarType, match, ...rest } = props;
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/GroupPage/GroupPage.js:9]: @Niknikson Are you sure that you need all rest props?

File: frontend/src/components/GroupSchedulePage/GroupSchedulePageTop/GroupSchedulePageTop.scss [hunk 28]
```diff
@@ -48,3 +48,7 @@ h1 {
     background: #d4dafd;
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/GroupSchedulePage/GroupSchedulePageTop/GroupSchedulePageTop.scss:48]: add to color constant

File: frontend/src/components/LessonsPage/LessonForm/LessonForm.js [hunk 29]
```diff
@@ -69,7 +69,8 @@ let LessonForm = (props) => {
         initialize,
         change,
         lessonTypes,
-        onSetSelectedCard,
+        selectLessonCard,
+        setUniqueError,
     } = props;
 
     const classes = useStyles();
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/LessonsPage/LessonForm/LessonForm.js:76]: @Yashchemskyi-Oleksandr change to scss

File: frontend/src/components/LessonsPage/LessonPage.js (hunk 14 above)
Comment (ViktoriiaKosovan) [reply frontend/src/components/LessonsPage/LessonPage.js:121]: saga is in saga folder, semester.js file

File: frontend/src/components/LoginForm/LoginForm.js [hunk 30]
```diff
@@ -1,82 +1,133 @@
...
+        setGroupId(0);
+        setTypeOfSchedule('full');
     });
-    const { t } = useTranslation('formElements');
-    const { handleSubmit, loginError, translation, setError, isLoading } = props;
-
-    const error = loginError;
-
+    const isValidForm = (formValues) => {
+        if (!formValues.email || !formValues.password) {
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/LoginForm/LoginForm.js:58]: Could you destructuring formValues?

File: frontend/src/components/Navigation/index.js [hunk 31]
```diff
@@ -0,0 +1,3 @@
+import NavigationPanel from './NavigationPanel';
+
+export default NavigationPanel;
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/Navigation/index.js]: Why we need this index.js?

File: frontend/src/components/Semester/SemesterCopyForm/SemesterCopyForm.js [hunk 32]
```diff
@@ -1,25 +1,18 @@
...
 import {
     FORM_SEMESTER_LABEL,
     FORM_SEMESTER_COPY_RED_HINT,
     FORM_SEMESTER_COPY_HINT,
-} from '../../constants/translationLabels/formElements';
-import { TEACHER_SCHEDULE_LABEL } from '../../constants/translationLabels/common';
+} from '../../../constants/translationLabels/formElements';
+import { TEACHER_SCHEDULE_LABEL } from '../../../constants/translationLabels/common';
 
 const useStyles = makeStyles((theme) => ({
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/Semester/SemesterCopyForm/SemesterCopyForm.js:17]: @ViktoriiaKosovan  change to scss

File: frontend/src/components/Semester/SemesterForm/SemesterForm.js [hunk 33]
```diff
@@ -0,0 +1,274 @@
...
+    const handleChange = (event, setState) => setState(event.target.checked);
+
+    const resetSemesterForm = () => {
+        setSelectedGroups([]);
+        clearSemesterService();
+    };
+
+    return (
+        <Card additionClassName="form-card semester-form">
+            <h2 style={{ textAlign: 'center' }}>
```
Comment (ViktoriiaKosovan) [inline frontend/src/components/Semester/SemesterForm/SemesterForm.js:145]: @ViktoriiaKosovan move to scss file

File: frontend/src/containers/AuthContainer/Logout/Logout.js [hunk 34]
```diff
@@ -0,0 +1,20 @@
+import React, { useEffect } from 'react';
+import { Redirect } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+import { logout } from '../../../actions/index';
+
+const Logout = (props) => {
```
Comment (ViktoriiaKosovan) [inline frontend/src/containers/AuthContainer/Logout/Logout.js]: @HiroFive Maybe it should be in component

File: frontend/src/helper/semesterFormValueMapper.js [hunk 35]
```diff
@@ -0,0 +1,38 @@
+import { get } from 'lodash';
+import { store } from '../store';
+
+export const semesterFormValueMapper = (semester) => {
+    const semesterDays = [];
+    const semesterClasses = [];
+    Object.keys(semester).forEach((prop) => {
+        if (get(semester, prop)) {
```
Comment (ViktoriiaKosovan) [inline frontend/src/helper/semesterFormValueMapper.js:8]: @ViktoriiaKosovan delete if (get(semester, prop))

File: frontend/src/actions/groups.js (hunk 2 above)
Comment (Niknikson) [reply frontend/src/actions/groups.js:10]: > @Niknikson Susses -> Success

fixed

File: frontend/src/actions/groups.js (hunk 3 above)
Comment (Niknikson) [reply frontend/src/actions/groups.js:24]: > @Niknikson as above

fixed

File: frontend/src/actions/groups.js (hunk 4 above)
Comment (Niknikson) [reply frontend/src/actions/groups.js:38]: > @Niknikson as above

fixed

File: frontend/src/components/GroupSidebar/GroupSidebar.js (hunk 11 above)
Comment (Niknikson) [reply frontend/src/components/GroupSidebar/GroupSidebar.js:15]: > @Niknikson why should we reassign this?

fixed

File: frontend/src/components/GroupSidebar/GroupSidebar.js (hunk 12 above)
Comment (Niknikson) [reply frontend/src/components/GroupSidebar/GroupSidebar.js:25]: > @Niknikson we can call this method toggleDisabled

fixed

File: frontend/src/components/GroupSidebar/GroupSidebar.js (hunk 13 above)
Comment (Niknikson) [reply frontend/src/components/GroupSidebar/GroupSidebar.js:31]: > @Niknikson arrow function wrapper is not necessary here

fixed

File: frontend/src/components/LessonsPage/LessonPage.js (hunk 14 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/components/LessonsPage/LessonPage.js:121]: resolve

File: frontend/src/components/LessonsPage/Search/Search.js (hunk 16 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/components/LessonsPage/Search/Search.js:40]: resolve

File: frontend/src/components/LessonsPage/LessonForm/LessonForm.js (hunk 29 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/components/LessonsPage/LessonForm/LessonForm.js:76]: resolve

File: frontend/src/components/Navigation/index.js (hunk 31 above)
Comment (HiroFive) [reply frontend/src/components/Navigation/index.js]: removed useless index file

File: frontend/src/containers/AuthContainer/Logout/Logout.js (hunk 34 above)
Comment (HiroFive) [reply frontend/src/containers/AuthContainer/Logout/Logout.js]: moved to components

File: frontend/public/index.html (hunk 1 above)
Comment (HiroFive) [reply frontend/public/index.html:41]: will be  removed

**Review by Yashchemskyi-Oleksandr — APPROVED** [review]

File: frontend/src/sagas/student.js [hunk 36]
```diff
@@ -0,0 +1,81 @@
...
+function* updateStudentWorker({ data }) {
+    try {
+        const res = yield call(axiosCall, STUDENT_URL, 'PUT', data);
+        yield put(updateStudent(res.data));
+        yield put(setStudent(null));
+        yield put(reset(STUDENT_FORM));
+        successHandler(
+            i18n.t(BACK_END_SUCCESS_OPERATION, {
+                cardType: i18n.t(FORM_STUDENT_LABEL),
+                actionType: i18n.t(UPDATED_LABEL),
```
Comment (Niknikson) [inline frontend/src/sagas/student.js:52]: refactor

File: frontend/src/sagas/semesters.js [hunk 37]
```diff
@@ -0,0 +1,301 @@
...
+import {
+    FORM_LESSON_LABEL,
+    FORM_SEMESTER_LABEL,
+} from '../constants/translationLabels/formElements';
+import { SEMESTER_FORM } from '../constants/reduxForms';
+
+export function* getAllSemestersItems() {
+    try {
+        const response = yield call(axiosCall, SEMESTERS_URL);
+        yield put(showAllSemesters(response.data.sort((a, b) => b.year - a.year)));
```
Comment (Niknikson) [inline frontend/src/sagas/semesters.js:46]: can you move sort in reducer ?

File: frontend/src/sagas/semesters.js [hunk 38]
```diff
@@ -0,0 +1,301 @@
...
+            ? i18n.t(error.response.data.message, error.response.data.message)
+            : 'Error';
+        const isOpen = true;
+        const type = snackbarTypes.ERROR;
+        yield put(setOpenSnackbar({ isOpen, type, message }));
+    }
+}
+export function* setGroupsToSemester({ semesterId, groups }) {
+    try {
+        const groupIds = groups.map((item) => `groupId=${item.id}`).join('&');
```
Comment (Niknikson) [inline frontend/src/sagas/semesters.js:84]: @ViktoriiaKosovan 
we can get groups from store direct

File: frontend/src/sagas/semesters.js [hunk 39]
```diff
@@ -0,0 +1,301 @@
...
+        const requestUrl = `${SEMESTERS_URL}/${semesterId}/groups?${groupIds}`;
+        const response = yield call(axiosCall, requestUrl, 'PUT');
```
Comment (Niknikson) [inline frontend/src/sagas/semesters.js:86]: @ViktoriiaKosovan 
create constants for PUT,POST...

File: frontend/src/sagas/schedule.js [hunk 40]
```diff
@@ -0,0 +1,441 @@
...
+        const isOpen = true;
+        const type = snackbarTypes.ERROR;
+        yield put(setOpenSnackbar({ isOpen, type, message }));
+        yield put(setLoading(false));
+    }
+}
+
+export function* addItemsToSchedule({ item }) {
+    try {
+        yield call(axiosCall, SCHEDULE_ITEMS_URL, 'POST', item);
```
Comment (Niknikson) [inline frontend/src/sagas/schedule.js:100]: @Akr1s 
maybe to create constants for POST...

File: frontend/src/sagas/schedule.js [hunk 41]
```diff
@@ -0,0 +1,441 @@
...
+        yield put(checkAvailabilitySchedule(response.data));
+        yield put(setLoading(false));
+    } catch (error) {
+        const message = error.response
+            ? i18n.t(error.response.data.message, error.response.data.message)
+            : 'Error';
+        const isOpen = true;
+        const type = snackbarTypes.ERROR;
+        yield put(setOpenSnackbar({ isOpen, type, message }));
+        yield put(setLoading(false));
```
Comment (Niknikson) [inline frontend/src/sagas/schedule.js:150]: @Akr1s 
maybe to use finally for yield put(setLoading(false));

File: frontend/src/services/groupService.js (hunk 19 above)
Comment (Niknikson) [reply frontend/src/services/groupService.js:37]: we will delete service

> replace `dispatch` with `mapDispatchToProps`

we will delete service

File: frontend/src/components/AddStudentForm/AddStudentForm.js (hunk 20 above)
Comment (Niknikson) [reply frontend/src/components/AddStudentForm/AddStudentForm.js:45]: > @Niknikson add to translation constants

fixed

File: frontend/src/components/AddStudentForm/AddStudentForm.js (hunk 21 above)
Comment (Niknikson) [reply frontend/src/components/AddStudentForm/AddStudentForm.js:106]: fixed

File: frontend/src/components/AddStudentForm/AddStudentForm.js (hunk 22 above)
Comment (Niknikson) [reply frontend/src/components/AddStudentForm/AddStudentForm.js:114]: fixed

File: frontend/src/components/AddStudentForm/AddStudentForm.js (hunk 23 above)
Comment (Niknikson) [reply frontend/src/components/AddStudentForm/AddStudentForm.js:56]: fixed

File: frontend/src/components/AddStudentForm/AddStudentForm.js (hunk 24 above)
Comment (Niknikson) [reply frontend/src/components/AddStudentForm/AddStudentForm.js:67]: fixed

File: frontend/src/components/Semester/SemesterForm/SemesterForm.scss (hunk 17 above)
Comment (ViktoriiaKosovan) [reply frontend/src/components/Semester/SemesterForm/SemesterForm.scss:43]: fixed

**Review by HiroFive — APPROVED** [review]

File: frontend/src/components/Semester/SemesterCopyForm/SemesterCopyForm.js (hunk 32 above)
Comment (ViktoriiaKosovan) [reply frontend/src/components/Semester/SemesterCopyForm/SemesterCopyForm.js:17]: fixed

File: frontend/src/components/Semester/SemesterItem.scss (hunk 18 above)
Comment (ViktoriiaKosovan) [reply frontend/src/components/Semester/SemesterItem.scss:32]: it will resolve

File: frontend/src/helper/semesterFormValueMapper.js (hunk 35 above)
Comment (ViktoriiaKosovan) [reply frontend/src/helper/semesterFormValueMapper.js:8]: fixed

File: frontend/src/components/Semester/SemesterForm/SemesterForm.js (hunk 33 above)
Comment (ViktoriiaKosovan) [reply frontend/src/components/Semester/SemesterForm/SemesterForm.js:145]: fixed

File: frontend/src/sagas/semesters.js (hunk 38 above)
Comment (ViktoriiaKosovan) [reply frontend/src/sagas/semesters.js:84]: it's a different groups and it isn't static

File: frontend/src/sagas/semesters.js (hunk 37 above)
Comment (ViktoriiaKosovan) [reply frontend/src/sagas/semesters.js:46]: fixed

File: frontend/src/sagas/schedule.js (hunk 40 above)
Comment (Akr1s) [reply frontend/src/sagas/schedule.js:100]: created

File: frontend/src/sagas/schedule.js (hunk 41 above)
Comment (Akr1s) [reply frontend/src/sagas/schedule.js:150]: changed in another branch

File: frontend/src/components/LessonsPage/Lessons.js (hunk 15 above)
Comment (Yashchemskyi-Oleksandr) [reply frontend/src/components/LessonsPage/Lessons.js:20]: resolve

File: frontend/src/components/GroupSchedulePage/GroupSchedulePageTop/GroupSchedulePageTop.scss (hunk 28 above)
Comment (Akr1s) [reply frontend/src/components/GroupSchedulePage/GroupSchedulePageTop/GroupSchedulePageTop.scss:48]: added

**Review by ViktoriiaKosovan — APPROVED** [review]

**Review by andrii-kuzmichov — APPROVED** [review]

File: frontend/src/components/GroupPage/GroupPage.scss (hunk 8 above)
Comment (Niknikson) [reply frontend/src/components/GroupPage/GroupPage.scss:42]: > Why exactly `12` and `13`?

12*10 is size for group card

**Pull request merged**